app/models/supplier.rb

```diff
@@ -13,4 +13,8 @@ class Supplier < ApplicationRecord
   }
 
   scope :excluding, ->(suppliers) { where.not(id: suppliers) }
+
+  def self.search(query)
+    query.blank? ? all : where('name ILIKE :query', query: "%#{query}%")
```

Comment:
I might be mistaken, but would `where(nil)` make this chainable vs `all` ?

Reply:
`all` is also chainable as it returns a relation:

```ruby
> Supplier.all.class
=> Supplier::ActiveRecord_Relation
```